src/realm/object-store/sync/sync_session.cpp

```diff
@@ -103,6 +103,27 @@ void SyncSession::become_active()
     }
 }
 
+void SyncSession::restart_session()
+{
+    util::CheckedLockGuard lock(m_state_mutex);
+    // Don't update the state if it is currently paused
+    if (m_state != State::Paused)
```

Comment:
If m_state == State::Paused, can't we just return early here?

Reply:
Yes, since the paused state infers that the session is not active, so I think that will be fine.